### PR TITLE
feat: replace QA_ENV with TWO_FA_ENABLED and remove demo scope

### PR DESCRIPTION
**Pull request by Nil20**
## Description

Clearly describe what has been changed. Include relevant context or background.
Explain how the issue was fixed (if applicable) and the root cause.

Link this pull request to the GitHub issue (and optionally name the branch `ocrvs-<issue #>`)

## Checklist

- [ ] I have linked the correct Github issue under "Development"
- [ ] I have tested the changes locally, and written appropriate tests
- [ ] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [ ] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: infrastructure/docker-compose.staging-deploy.yml [hunk 1]
```diff
@@ -32,23 +32,20 @@ services:
...
       - SENTRY_DSN=${SENTRY_DSN:-}
     deploy:
       replicas: 1
 
   user-mgnt:
     environment:
-      - QA_ENV=true
       - NODE_ENV=production
       - SENTRY_DSN=${SENTRY_DSN:-}
       - MONGO_URL=mongodb://user-mgnt:${USER_MGNT_MONGODB_PASSWORD}@mongo1/user-mgnt?replicaSet=rs0
```
Comment (Nil20) [inline infrastructure/docker-compose.staging-deploy.yml:51]: `TWO_FA_ENABLED` is not set explicitly in `docker-compose.staging-deploy.yml` — the default value of true in both auth and gateway ensures 2FA is enabled in staging without needing an       
  override.

**Review by rikukissa — APPROVED** [review]

**Comment by Nil20** [comment]
Successful e2e run: https://github.com/opencrvs/e2e/actions/runs/24140464912
